### ISSUE TITLE
Documentation on how to build librsync under/for Windows

### ISSUE THREAD

**Issue by ericzolf** [librsync/librsync#171]
I've started to work on this for the rdiff-backup project at https://github.com/rdiff-backup/rdiff-backup/pull/131 but I'm struggling, not being knowledgeable neither in Windows nor in CMake, and would appreciate some support / documentation.

**Comment by ardovm** [comment]
Hello,
I have worked on the same goal, on my ''physical'' PC with Windows 10.

At the moment, the commands that seem to work are in my branch of rdiff-backup, in the file `dist_win32.py`:
```
cmake -A Win32 .
cmake --build . --config Release
cmake --install . --config Release
```
(I added another flag to the first invocation of CMake to change the "installation" directory, but that is the core)

These commands must be run in a "Developer Command Prompt for VS 2019".

**Comment by dbaarda** [comment]
And if you want to build a static library instead of a dll, change the --build line to;

cmake --build . --config Release -D BUILD_SHARED_LIBS=OFF

Note this will only work with current master. I'm ready to release it as a new v2.2.0 as soon as you are happy with it.

**Comment by ericzolf** [comment]
As you can see from my PR, I've managed to build both librsync 2.0.2 from tarball and master from Git, and bind it with rdiff-backup. Thanks for the hints, they were much helpful!

Do you want to keep the issue open until a documentation has indeed be written? I would see it as helpful for others.

**Comment by dbaarda** [comment]
Yeah, i think we should leave this open till the documentation is updated and merged.

Pull requests are welcome.

**Comment by ericzolf** [comment]
I can document my learnings but:
1. they're limited to what I experienced (Mingw32 on Linux isn't part of it)
2. I would need instructions on where to put them, would `doc/build_windows.md` + adding link at the end of `README.md` be good?

**Comment by dbaarda** [comment]
Sorry for taking so long to reply. I'm planning to release v2.2.0 today and we can sort this out for the next release.

I can give more detailed advise on where/how to add docs after i formulate some thoughts about it.

An important thing to remember is whatever you can contribute, no matter how feeble or inadequate it seems, is still infinitely better than the nothing we have right now.

**Comment by mgrandi** [comment]
i can confirm that the steps here https://github.com/librsync/librsync/issues/171#issuecomment-537817885 seem to at least build the files correctly, i am having trouble getting the program that uses librsync to work so i can't verify if it functions correctly

**Comment by dbaarda** [comment]
Note that the github actions added in #225 now builds and checks rdiff on Windows using popt from the vcpkg repo. You can look at the check.yml file for how to fetch/install the package, and the cmake cmdlines to build it.
